exasol_integration_test_docker_environment/lib/test_environment/ports.py

```diff
@@ -0,0 +1,55 @@
+import socket
+from contextlib import ExitStack
+from typing import List, Optional
+
+
+def find_free_ports(num_ports: int) -> List[int]:
+    def new_socket():
+        return socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    def bind(sock: socket.socket, port: int):
+        sock.bind(('', port))
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    def acquire_port_numbers(num_ports: int) -> List[int]:
+        with ExitStack() as stack:
+            sockets = [stack.enter_context(new_socket()) for dummy in range(num_ports)]
+            for sock in sockets:
+                bind(sock, 0)
+                yield sock.getsockname()[1]
+    def check_port_numbers(ports):
+        with ExitStack() as stack:
+            sockets_and_ports = [(stack.enter_context(new_socket()), port) for port in ports]
+            for sock, port in sockets_and_ports:
+                bind(sock, port)
+    ports = list(acquire_port_numbers(num_ports))
+    check_port_numbers(ports)
+    return ports
+
+
+class PortsType(type):
```

Comment:
If I understand correct, you want these properties as read only class attributes. Please add this to the docstring.

Reply:
will do so.